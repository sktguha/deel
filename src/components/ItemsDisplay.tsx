import { genericFn } from "../types/Common.types";
import { MenuItem } from "../types/Common.types";
import { DisplayItem } from "./DisplayItem";

interface ItemsDisplayPropTypes {
  itemsRef: any;
  availableOptions: MenuItem[];
  maxDisplayItems: number;
  selectedOptions: MenuItem[];
  onSelectedOptionsChange: genericFn;
  getBoldedText: genericFn;
  inputRef: any;
}

export function ItemsDisplay({
  itemsRef, availableOptions, maxDisplayItems, selectedOptions, onSelectedOptionsChange, getBoldedText,
  inputRef }: ItemsDisplayPropTypes) {
  const availableOptionsLimit = availableOptions.slice(0, maxDisplayItems);
  return (<div className="available-options-to-select" ref={itemsRef}>
    {
      availableOptionsLimit.map(
        (option, i) => {
          // alert(i);
          return (
            <DisplayItem isLast={i === availableOptionsLimit.length-1} key={option.id} {...{ selectedOptions, option, onSelectedOptionsChange, getBoldedText, inputRef }} />
          );
        }
      )
    }
  </div>);
}
