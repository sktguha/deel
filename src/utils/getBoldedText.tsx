export default function getBoldedText(label, text) {
    return <div dangerouslySetInnerHTML={{
      __html: label.split(text).join("<b>" + text + "</b>")
    }} />;
  }