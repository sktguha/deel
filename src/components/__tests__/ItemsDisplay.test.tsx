import React from 'react';
import { ItemsDisplay } from "../ItemsDisplay"
import { render, screen } from '@testing-library/react';
import getBoldedText from '../../utils/getBoldedText';


test("ItemsDisplay base", () => {
  const { container } = render(<ItemsDisplay getBoldedText={getBoldedText} inputRef={{ current: { value: "te" } }} availableOptions={[{ id: 1, label: "test" }]} />)
  expect(container).toMatchSnapshot();
})

test("ItemsDisplay multiple options", () => {
  const { container } = render(<ItemsDisplay getBoldedText={getBoldedText} inputRef={{ current: { value: "te" } }}
    availableOptions={[
      { id: 1, label: "test" },
      { id: 2, label: "test2" }]
    } />)
  expect(container).toMatchSnapshot();
})