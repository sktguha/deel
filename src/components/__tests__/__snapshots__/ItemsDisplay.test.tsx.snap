// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsDisplay base 1`] = `
<div>
  <div
    class="available-options-to-select"
  >
    <div
      class="menu-item menu-item-last"
      style="cursor: pointer;"
    >
      <div>
        <b>
          te
        </b>
        st
      </div>
    </div>
  </div>
</div>
`;

exports[`ItemsDisplay multiple options 1`] = `
<div>
  <div
    class="available-options-to-select"
  >
    <div
      class="menu-item false"
      style="cursor: pointer;"
    >
      <div>
        <b>
          te
        </b>
        st
      </div>
    </div>
    <div
      class="menu-item menu-item-last"
      style="cursor: pointer;"
    >
      <div>
        <b>
          te
        </b>
        st2
      </div>
    </div>
  </div>
</div>
`;
